lib/smart_app_launch/discovery_stu2_2_group.rb

```diff
@@ -0,0 +1,191 @@
+    short_description 'Retrieve server\'s SMART on FHIR configuration.'
+    description %(
+      # Background
+
+      The #{title} Sequence test looks for authorization endpoints and SMART
+      capabilities as described by the [SMART App Launch
+      Framework](http://hl7.org/fhir/smart-app-launch/STU2.2/).
+      The SMART launch framework uses OAuth 2.0 to *authorize* apps, like
+      Inferno, to access certain information on a FHIR server. The
+      authorization service accessed at the endpoint allows users to give
+      these apps permission without sharing their credentials with the
+      application itself. Instead, the application receives an access token
+      which allows it to access resources on the server. The access token
+      itself has a limited lifetime and permission scopes associated with it.
+      A refresh token may also be provided to the application in order to
+      obtain another access token. Unlike access tokens, a refresh token is
+      not shared with the resource server. If OpenID Connect is used, an id
+      token may be provided as well. The id token can be used to
+      *authenticate* the user. The id token is digitally signed and allows the
+      identity of the user to be verified.
+
+      # Test Methodology
+
+      This test suite will examine the SMART on FHIR configuration contained
+      in the `/.well-known/smart-configuration` endpoint.
+
+      For more information see:
+
+      * [SMART App Launch Framework](http://hl7.org/fhir/smart-app-launch/STU2.2/)
+      * [The OAuth 2.0 Authorization Framework](https://tools.ietf.org/html/rfc6749)
+      * [OpenID Connect Core](https://openid.net/specs/openid-connect-core-1_0.html)
+    )
+
+    test from: :well_known_endpoint_stu2_2,
+         config: {
+           outputs: {
+             well_known_authorization_url: { name: :smart_authorization_url },
+             well_known_token_url: { name: :smart_token_url }
+           }
+         }
+    test from: :well_known_capabilities_stu2
+
+    test do
+      include URLHelpers
+
+      title 'Conformance/CapabilityStatement provides OAuth 2.0 endpoints'
+      description %(
+        If a server requires SMART on FHIR authorization for access, its
+        metadata must support automated discovery of OAuth2 endpoints.
+
+        For requests from any origin, CORS configuration permits access to the public discovery endpoints
+        (.well-known/smart-configuration and metadata). This test verifies that the metadata
+        request is returned with the appropriate CORS header.
+      )
+      input :url
+      output :capability_authorization_url,
+             :capability_introspection_url,
+             :capability_management_url,
+             :capability_registration_url,
+             :capability_revocation_url,
+             :capability_token_url
+
+      fhir_client do
+        url :url
+        headers 'Origin' => Inferno::Application['base_url']
```

Comment:
This should only be scheme and host name (and port of using a non-standard port). That is `inferno_host` rather than `base_url`.